### PR TITLE
[AGENT] Reduce memory consumption for Option fields

### PR DESCRIPTION
**Pull request by rvql**
### This PR is for:
- Agent

### Reduce memory consumption for Option fields

indentation means fields in struct
`->` means boxed
`2x` means two instances
`+-`  means types in enum
`+>`  means boxed types in enum

#### Before
```
crate::flow_generator::flow_node::FlowNode: 2328
    crate::common::TaggedFlow: 696
        crate::common::flow::Flow: 632
            crate::common::flow::FlowKey: 72
         2x crate::common::flow::FlowMetricsPeer: 120
            crate::common::flow::TunnelField: 44
            crate::common::flow::FlowPerfStats: 144
        crate::common::tag::Tag: 64
    crate::flow_generator::flow_state::FlowState: 1
    crate::flow_generator::perf::FlowPerf: 1472
        crate::flow_generator::perf::L4FlowPerfTable: 752
         -- crate::flow_generator::perf::tcp::TcpPerf: 752
         |      crate::flow_generator::perf::tcp::PerfControl: 384
         |       2x crate::flow_generator::perf::tcp::SessionPeer: 192
         |      crate::flow_generator::perf::tcp::PerfData: 352
         -- crate::flow_generator::perf::udp::UdpPerf: 56
        crate::flow_generator::perf::L7FlowPerfTable: 344
         +- crate::flow_generator::perf::dns::DnsPerfData: 96
         +- crate::flow_generator::perf::mq::KafkaPerfData: 104
         +- crate::flow_generator::perf::mq::MqttPerfData: 96
         +- crate::flow_generator::perf::sql::RedisPerfData: 96
         +- crate::flow_generator::perf::rpc::DubboPerfData: 112
         +- crate::flow_generator::perf::sql::MysqlPerfData: 104
         +- crate::flow_generator::perf::http::HttpPerfData: 112
         +- crate::flow_generator::protocol_logs::sql::PostgresqlLog: 288
         +- crate::flow_generator::protocol_logs::rpc::ProtobufRpcWrapLog: 288
         +- crate::flow_generator::protocol_logs::rpc::SofaRpcLog: 336
        crate::common::l7_protocol_log::L7ProtocolParser: 344
         +- crate::flow_generator::protocol_logs::http::HttpLog: 320
         +- crate::flow_generator::protocol_logs::dns::DnsLog: 88
         +- crate::flow_generator::protocol_logs::rpc::ProtobufRpcWrapLog: 288
         +- crate::flow_generator::protocol_logs::rpc::SofaRpcLog: 336
         +- crate::flow_generator::protocol_logs::sql::MysqlLog: 128
         +- crate::flow_generator::protocol_logs::mq::KafkaLog: 96
         +- crate::flow_generator::protocol_logs::sql::RedisLog: 160
         +- crate::flow_generator::protocol_logs::sql::PostgresqlLog: 288
         +- crate::flow_generator::protocol_logs::rpc::DubboLog: 200
         +- crate::flow_generator::protocol_logs::mq::MqttLog: 96
 2x npb_pcap_policy::PolicyData: 32
 2x crate::common::endpoint::EndpointData: 64
    packet_sequence_block::PacketSequenceBlock: 0
```

#### After
```
crate::flow_generator::flow_node::FlowNode: 664
    crate::common::TaggedFlow: 496
        crate::common::flow::Flow: 432
            crate::common::flow::FlowKey: 72
         2x crate::common::flow::FlowMetricsPeer: 104
            crate::common::flow::TunnelField: 44
         -> crate::common::flow::FlowPerfStats: 144
        crate::common::tag::Tag: 64
    crate::flow_generator::flow_state::FlowState: 1
 -> crate::flow_generator::perf::FlowPerf: 80
        crate::flow_generator::perf::L4FlowPerfTable: 32
         +> crate::flow_generator::perf::tcp::TcpPerf: 600
         |      crate::flow_generator::perf::tcp::PerfControl: 352
         |       2x crate::flow_generator::perf::tcp::SessionPeer: 176
         |      crate::flow_generator::perf::tcp::PerfData: 232
         -- crate::flow_generator::perf::udp::UdpPerf: 32
     -> crate::flow_generator::perf::L7FlowPerfTable: 120
         +- crate::flow_generator::perf::dns::DnsPerfData: 96
         +- crate::flow_generator::perf::mq::KafkaPerfData: 104
         +- crate::flow_generator::perf::mq::MqttPerfData: 96
         +- crate::flow_generator::perf::sql::RedisPerfData: 96
         +- crate::flow_generator::perf::rpc::DubboPerfData: 112
         +- crate::flow_generator::perf::sql::MysqlPerfData: 104
         +- crate::flow_generator::perf::http::HttpPerfData: 112
         +> crate::flow_generator::protocol_logs::sql::PostgresqlLog: 288
         +> crate::flow_generator::protocol_logs::rpc::ProtobufRpcWrapLog: 288
         +> crate::flow_generator::protocol_logs::rpc::SofaRpcLog: 336
     -> crate::common::l7_protocol_log::L7ProtocolParser: 128
         +- crate::flow_generator::protocol_logs::http::HttpLog: 320
         +- crate::flow_generator::protocol_logs::dns::DnsLog: 88
         +- crate::flow_generator::protocol_logs::rpc::ProtobufRpcWrapLog: 288
         +- crate::flow_generator::protocol_logs::rpc::SofaRpcLog: 336
         +- crate::flow_generator::protocol_logs::sql::MysqlLog: 128
         +- crate::flow_generator::protocol_logs::mq::KafkaLog: 96
         +- crate::flow_generator::protocol_logs::sql::RedisLog: 160
         +- crate::flow_generator::protocol_logs::sql::PostgresqlLog: 288
         +- crate::flow_generator::protocol_logs::rpc::DubboLog: 200
         +- crate::flow_generator::protocol_logs::mq::MqttLog: 96
 2x npb_pcap_policy::PolicyData: 32
 2x crate::common::endpoint::EndpointData: 64
 -> packet_sequence_block::PacketSequenceBlock: 0
```

### CODE REVIEW

File: agent/src/flow_generator/perf/mod.rs [hunk 1]
```diff
@@ -90,33 +89,165 @@ pub trait L7FlowPerf {
...
+            Self::Dubbo(p) => p.data_updated(),
+            Self::Mysql(p) => p.data_updated(),
+            Self::Http(p) => p.data_updated(),
+            Self::Postgresql(p) => p.data_updated(),
+            Self::ProtobufRpc(p) => p.data_updated(),
+            Self::SofaRpc(p) => p.data_updated(),
+        }
+    }
+}
+*/
```
Comment (sharang) [inline agent/src/flow_generator/perf/mod.rs:242]: 这个还要吗

Comment (rvql) [reply agent/src/flow_generator/perf/mod.rs:242]: 不要了

**Comment by rvql** [comment]
还有个seg fault

File: agent/src/common/timestamp.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
+/*
```
Comment (rvql) [inline agent/src/common/timestamp.rs:1]: 还有个segfault待解决

**Review by sharang — DISMISSED** [review]

**Review by sharang — APPROVED** [review]

**Pull request merged**